File: RemoveRecursionRefactoring/src/ro/pub/cs/diploma/RemoveRecursionBundle.java
package ro.pub.cs.diploma;

import com.intellij.CommonBundle;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.PropertyKey;

import java.lang.ref.Reference;
import java.lang.ref.SoftReference;
import java.util.ResourceBundle;

/**
 * @see com.siyeh.InspectionGadgetsBundle
 */
class RemoveRecursionBundle {
  @NotNull
  static String message(@NotNull @PropertyKey(resourceBundle = BUNDLE) String key, @NotNull Object... params) {
    return CommonBundle.message(getBundle(), key, params);
  }

  private static Reference<ResourceBundle> ourBundle;
  @NonNls private static final String BUNDLE = "ro.pub.cs.diploma.RemoveRecursionBundle";

  private RemoveRecursionBundle() {
  }

  private static ResourceBundle getBundle() {
    ResourceBundle bundle = com.intellij.reference.SoftReference.dereference(ourBundle);
    if (bundle == null) {
      bundle = ResourceBundle.getBundle(BUNDLE);
      ourBundle = new SoftReference<>(bundle);
    }
    return bundle;
  }
}

File: inspection/src/main/kotlin/ro/pub/cs/diploma/RemoveRecursionBundle.kt
package ro.pub.cs.diploma

import com.intellij.AbstractBundle
import org.jetbrains.annotations.NonNls
import org.jetbrains.annotations.PropertyKey
import java.util.ResourceBundle

/**
 * @see com.siyeh.InspectionGadgetsBundle
 */
object RemoveRecursionBundle {
    @NonNls private const val BUNDLE = "RemoveRecursionBundle"

    private val bundle: ResourceBundle by lazy(LazyThreadSafetyMode.NONE) { ResourceBundle.getBundle(BUNDLE) }

    fun message(
        @PropertyKey(resourceBundle = BUNDLE) key: String,
        vararg params: Any,
    ): String = AbstractBundle.message(bundle, key, *params)
}
